### DISCUSSION TITLE
Identify the user in a pre save triggered script.

### DISCUSSION THREAD

**Discussion by RolandvonBorstel** [sadieblue/agilityblue.workspace-scripts#33] [Q&A] [answered]
Is it possible to identify the user that triggers a pre-save script to only take actions for specific users but not all users?

The LastUpdatedBy fields seem to only get updated when the object is actually saved.

**Comment by mrabb24** [comment, marked as answer]
Hello,

There is a special variable that is injected for events that you can use which holds the id and display name information of the user who triggers the event:

```powershell
if ($null -ne $ABVAREvent)
{
    $eventTriggeredBy = $ABVAREvent.CreatedBy.ToString() | ConvertFrom-Json
    
    $triggeredByUserId = $eventTriggeredBy.Id
    $triggeredByUserName = $eventTriggeredBy.DisplayValue
    
    Write-Output "This event was triggered by $($triggeredByUserName) (User Id: $($triggeredByUserId))"
}

return (Get-InboundObjectInstance)
```

This would produce something like the following in a pre-save script:

```
This event was triggered by Matthew Rabb (User Id: 25a34a2f-e3d8-4690-88f5-6b399cf88c4c)
AgilityBlue.ViewModels.Workspace.MediaLogEntryViewModel
```

Matt
